src/second_ode_solve.jl

```diff
@@ -0,0 +1,63 @@
+using Flux
+struct NNODE2{C,O} <: NeuralNetDiffEqAlgorithm
+    chain::C
+    opt::O
+end
+NNODE2(chain;opt=Flux.ADAM(0.1)) = NNODE2(chain,opt)
+
+function DiffEqBase.solve(
+    prob::DiffEqBase.AbstractODEProblem,
+    alg::NeuralNetDiffEqAlgorithm,
+    args...;
+    dt,
+    timeseries_errors = true,
+    save_everystep=true,
+    adaptive=false,
+    abstol = 1f-6,
+    verbose = false,
+    maxiters = 100)
+
+    DiffEqBase.isinplace(prob) && error("Only out-of-place methods are allowed!")
+
+    u0, du0 = prob.u0
+    tspan = prob.tspan
+    f = prob.f
+    p = prob.p
+
+    #hidden layer
+    chain  = alg.chain
+    opt    = alg.opt
+    ps     = Flux.params(chain)
+    data   = Iterators.repeated((), maxiters)
+
+    #train points generation
+    ts = []
+    cnt = 0.0f0
+    for ii in 1:Int((tspan[2]-tspan[1])/dt[1]+1)
+        push!(ts, cnt)
+        cnt += dt[1]
+    end
+    
+    #Initial Value Problem
+    phi(t) = (u0 .+ du0 .* (t .- tspan[1]) .+ (t .- tspan[1]) .^ 2 .* chain([t]))[1]
+
+
+    dfdx = t -> (phi(t+sqrt(eps(typeof(dt[1])))) - phi(t)) / sqrt(eps(typeof(dt[1])))
+    g(tp) = sum(dfdx(tp))
```

Comment:
I don't think this is right because of the `sum`, nor a necessarily good way to do it. I would directly calculate the second derivative, i.e. 

```julia
epsilon = cbrt(eps(real(T)))
(phi(t+epsilon) - 2phi(t) + phi(t-epsilon))/epsilon^2
```

https://en.wikipedia.org/wiki/Finite_difference where `epsilon = cbrt(eps(real(T)))` is taken from https://github.com/JuliaDiff/FiniteDiff.jl/blob/master/src/epsilons.jl#L29 as an optimal choice.

Reply:
I have briefly read through the links (will study them more later) and will amend the code!